### DISCUSSION TITLE
Is there any way to append an initial field?

### DISCUSSION THREAD

**Discussion by ag-gaphp** [MatthiasValvekens/pyHanko#420] [Q&A] [answered]
Adobe Acrobat and online eSign platforms have a distinction between two types of fields that can be signed: full signature and initials. Just curious if there is a way to specify a field to be an "initial" type when using `append_signature_field()`?

**Comment by MatthiasValvekens** [comment, marked as answer]
As far as I'm aware, that would be a proprietary extension used in the Adobe ecosystem.

Not sure I'd necessarily want to support this in-tree (maintaining proprietary integrations is usually not fun), but if you have some sample files, I could take a quick look to see if there's an easy workaround.

**Comment by ag-gaphp** [comment]
I appreciate the reply, my apologies for the delay. I hadn't considered that it might be something Adobe specific, but it makes sense and I understand not wanting to have to maintain it, especially with my experience this morning. I cannot, for the life of me, get a PDF to save with a proper initial field any more from Adobe Acrobat Pro. Every time, it just reverts to a plain old text field and has no esigning capabilities at all.

I'm baffled because there is a specific "Add Initial Field" button when preparing a form, and we have a bunch of files that had initial fields last week when I submitted this question. However, opening any of those files today only has plain text fields in place of where the initials were. I don't know if this is some recent change by Adobe or another coworker not speaking up, but any file in our bucket of forms are no longer showing "proper" initial fields... I have to go convert all of them to regular signature fields, so I guess my question is moot at this point.

I also tried to get a sample from DocuSign, but they don't allow you to download the unfilled PDFs. Has to be sent out for signing and completed before you can get the PDF. I attached one of these completed forms just in case it might provide some info, but I'm not sure if them already being signed affects what you are after or not.

No sweat if it's not possible! Having initial fields is mainly just an aesthetic thing, since I personally see no difference between full signatures and initials. Both only require the user to click the field and move on, so it doesn't save any time for our patients.

[DocuSign_initials_example.pdf](https://github.com/MatthiasValvekens/pyHanko/files/15041679/DocuSign_initials_example.pdf)

**Reply by MatthiasValvekens** [reply]
Yeah, that's more or less what I thought.

By the way, the DocuSign file you just uploaded contains only one "real" signature field. The others are not implemented as form fields at all. This is kinda par for the course with these eSigning platforms that "own" the full signing process start to finish--they don't have to care about standards beyond the bare minimum, because the walled-garden approach where they don't have to interoperate with anyone is their whole business model ;).

**Reply by ag-gaphp** [reply]
Interesting! I guess the initials are just a placebo and mean nothing, lmao. Thanks again for your time, and a huge thank you for the python module itself!
